src/systemctl/systemctl.c

```diff
@@ -2098,7 +2097,7 @@ static int set_default(int argc, char *argv[], void *userdata) {
                 if (r < 0)
                         return log_error_errno(r, "Failed to set default target: %s", bus_error_message(&error, r));
 
-                r = bus_deserialize_and_dump_unit_file_changes(reply, arg_quiet, NULL, NULL);
+                r = bus_deserialize_and_dump_unit_file_changes(reply, arg_quiet, &changes, &n_changes);
```

Comment:
Looks like we need some
```
unit_file_changes_free(changes, n_changes);
```
I will take a look tomorrow when I am back at physical keyboard.